src/network/mcpe/handler/PreSpawnPacketHandler.php

```diff
@@ -144,6 +150,11 @@ public function setUp() : void{
 			$this->session->getLogger()->debug("Sending creative inventory data");
 			$this->inventoryManager->syncCreative();
 
+			$this->session->getLogger()->debug("Sending armor trim data");
+			$patterns = array_map(fn(ArmorTrimPattern $pattern) => new TrimPattern($pattern->getItemId(), $pattern->value), ArmorTrimPattern::cases());
+			$materials = array_map(fn(ArmorTrimMaterial $material) => new TrimMaterial($material->value, $material->getColor(), $material->getItemId()), ArmorTrimMaterial::cases());
+			$this->session->sendDataPacket(TrimDataPacket::create($patterns, $materials));
```

Comment:
Is this customizable? If so, why are we using hard enums for this?

Reply:
@HimmelKreis4865 I don't know about patterns, but colours can be definitely customized. (For example, I can change the amethyst colour from purple to white). What do you think about allowing plugins to customize this?
![Screenshot_393](https://github.com/pmmp/PocketMine-MP/assets/63200545/5ba52c28-a641-48b4-b86e-340d63e0bb0c)

Reply:
yeah I'm aware of this, and this should be customizable but I did not yet find a solution for this problem that satisfies me

Reply:
I pushed a commit to make this all customizable, please tell me what you think of this

Reply:
> To me it looks good, but I'm not sure about passing type id and serialized item name to the constructor. Wouldn't a better to pass an Item and then serialize it when sending TrimData?
(from ipad54 on discord)
what do you guys think about this?

Reply:
See above.